src/Symfony/Component/Console/Helper/TreeBuilder.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class TreeBuilder
```

Comment:
Why not just one class? TreeNode is the 'root,' a self-nested class, and it could be responsible for instantiating itself with things like arrays or whatever. For example, TreeNode::buildYourselfWithAnArray (or just 'fromArray', this is just to illustrate the idea). This would be more fluent and intuitive. WYT ? 

In the TreeNode class :
```php
    public static function fromArray(array $array, ?string $root = null): TreeNode
    {
        $root = new self($root ?? '');
        $root->buildTree($root, $array); // Not static anymore

        return $root;
    }
```

Reply:
I don't know... as you see there is now a fromFinder method... i'm not sure the Tree should know anything about it

Reply:
Yeah, we could keep it to add more "strategy" from anything else that the FInder